### ISSUE TITLE
Drawer with tab controls should allow keyboard navigation

### ISSUE THREAD

**Issue by KonradWroblewski** [Sage/carbon#5597]
### Current behaviour

Keyboard navigation doesn't work properly in left sidebar. We can't navigate by TAB key or even arrow keys.

![image](https://user-images.githubusercontent.com/112560351/200289757-5a2dcfec-5a02-468c-871c-9af93fe9c9cc.png)

### Expected behaviour

User can navigate between tabs/menuItem by keyboard.

### CodeSandbox or Storybook URL

https://carbon.sage.com/?path=/story/drawer--with-tab-controls

### JIRA Ticket (Sage Only)

SBS-33896

### Carbon Version

111.8.0

### What browsers are you seeing the problem on?

Chrome

### What Operating System are you seeing the problem on?

Windows

### Anything else we should know?

The given SAGE JIRA Ticket is related to the focus (other issue ticket), but the navigation and focus in this case are related.

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.
